centreon/www/class/centreonLogAction.class.php

```diff
@@ -65,42 +65,48 @@ public function __construct($usr)
      *  Inserts configuration into DB
      */
 
-    public function insertFieldsNameValue($logId, $fields)
+    public function insertFieldsNameValue($logId, $fields): void
     {
         global $pearDBO;
 
-        $query = "INSERT INTO `log_action_modification` (field_name, field_value, action_log_id) VALUES ";
-        $append = "";
-        foreach ($fields as $key => $value) {
-            $query .= $append . "('" . CentreonDB::escape($key) . "', '" . CentreonDB::escape($value) . "', '" .
-                CentreonDB::escape($logId) . "')";
-            $append = ", ";
+        $statement = $pearDBO->prepare("INSERT INTO `log_action_modification` (field_name, field_value, action_log_id)
+                                            VALUES (:field_key, :field_value, :logId) ");
+        foreach ($fields as $field_key => $field_value) {
+            $key = CentreonDB::escape($field_key);
+            $value = CentreonDB::escape($field_value);
+            $statement->bindValue(':field_key', $key);
+            $statement->bindValue(':field_value', $value);
+            $statement->bindValue(':logId', $logId, PDO::PARAM_INT);
         }
```

Comment:
**Now**: You're inserting one line into DB and bind the last `$key` & `$value` of `$fields` array to the query
**Before**: The number of lines inserted into DB equals to the length of `$fields` array.

Please re-check your algorithm.